src/hotspot/share/opto/ifnode.cpp

```diff
@@ -859,7 +860,7 @@ bool IfNode::has_only_uncommon_traps(ProjNode* proj, ProjNode*& success, ProjNod
     }
 
     CallStaticJavaNode* unc = nullptr;
-    ProjNode* unc_proj = uncommon_trap_proj(unc);
+    IfProjNode* unc_proj = uncommon_trap_proj(unc)->as_IfProj();
```

Comment:
I think we can also make `uncommon_trap_proj` return an `IfProjNode`. But I want to tackle that separately.